server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java

```diff
@@ -248,69 +270,69 @@ private Table buildTable(RestRequest request, ClusterStateResponse state, Indice
-
-            table.addCell(commonStats == null ? null : commonStats.getMerge().getCurrent());
-            table.addCell(commonStats == null ? null : commonStats.getMerge().getCurrentNumDocs());
-            table.addCell(commonStats == null ? null : commonStats.getMerge().getCurrentSize());
-            table.addCell(commonStats == null ? null : commonStats.getMerge().getTotal());
-            table.addCell(commonStats == null ? null : commonStats.getMerge().getTotalNumDocs());
-            table.addCell(commonStats == null ? null : commonStats.getMerge().getTotalSize());
-            table.addCell(commonStats == null ? null : commonStats.getMerge().getTotalTime());
-
-            table.addCell(commonStats == null ? null : commonStats.getRefresh().getTotal());
-            table.addCell(commonStats == null ? null : commonStats.getRefresh().getTotalTime());
-            table.addCell(commonStats == null ? null : commonStats.getRefresh().getListeners());
-
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getFetchCurrent());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getFetchTime());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getFetchCount());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getOpenContexts());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getQueryCurrent());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getQueryTime());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getQueryCount());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getScrollCurrent());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getScrollTime());
-            table.addCell(commonStats == null ? null : commonStats.getSearch().getTotal().getScrollCount());
-
-            table.addCell(commonStats == null ? null : commonStats.getSegments().getCount());
-            table.addCell(commonStats == null ? null : commonStats.getSegments().getMemory());
-            table.addCell(commonStats == null ? null : commonStats.getSegments().getIndexWriterMemory());
-            table.addCell(commonStats == null ? null : commonStats.getSegments().getVersionMapMemory());
-            table.addCell(commonStats == null ? null : commonStats.getSegments().getBitsetMemory());
-
-            table.addCell(shardStats == null || shardStats.getSeqNoStats() == null ? null : shardStats.getSeqNoStats().getMaxSeqNo());
-            table.addCell(shardStats == null || shardStats.getSeqNoStats() == null ? null : shardStats.getSeqNoStats().getLocalCheckpoint());
-            table.addCell(commitStats == null || shardStats.getSeqNoStats() == null ? null : shardStats.getSeqNoStats().getGlobalCheckpoint());
```

Comment:
this seems to be a bug anyway since we checking `commitStats` and then access `shardStats`

Reply:
++